lib/fauxpaas/artifact.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Fauxpaas
+  class Artifact
+    def initialize(signature:, fs:)
+      dir = fs.mktmpdir
+      @fs = fs
+      @shared_path = extract_ref(signature.shared, dir/"shared")
+      @unshared_path = extract_ref(signature.unshared, dir/"unshared")
+      @source_path = extract_ref(signature.source, dir/"source")
+    end
+
+    def extract_ref(ref, path)
+      fs.mkdir_p path
+      add_reference(ref, path)
+      path
+    end
+
+    def add_reference(reference, base)
```

Comment:
Sort of unrelated to the refactoring, but `add_reference` seems pretty opaque to me. Maybe it's sort of "implementation speak" (scm terminology leaking in)? Or maybe it's the best name and a comment would give good context?